### PR TITLE
fix: icons for message-box

### PR DESCRIPTION
**Pull request by stefanoScalzo**
## Related Issue
Closes SAP/fundamental-styles#1896

## Description
Fixes the icons so that they are displayed with message box
### Before:
Screen Shot 2020-11-24 at 10 38 11 AM

### After:
Screen Shot 2020-11-24 at 10 37 51 AM

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: src/message-box.scss [hunk 1]
```diff
@@ -99,8 +100,8 @@ $message-box-states: (
         box-shadow: map_get($state-set, "boxShadow");
       }
 
-      .#{$block}__title {
-        @include fd-icon-base () {
+      .#{$message}__title {
```
Comment (InnaAtanasova) [inline src/message-box.scss:103]: Why are you bringing `fd-message` component here? It should be the way it was before.

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
I am pretty sure you are referencing the wrong issue in your PR template. You are referencing an issue about a border on the sub-header that shouldn't be there.

File: src/message-box.scss (hunk 1 above)
Comment (InnaAtanasova) [reply src/message-box.scss:103]: Also, we are no longer referencing other components this way. We rely on specificity.

Comment (stefanoScalzo) [reply src/message-box.scss]: fixed

File: src/message-box.scss [hunk 2]
```diff
@@ -5,6 +5,7 @@
 $block: #{$fd-namespace}-message-box;
 $bar: #{$fd-namespace}-bar;
 $button: #{$fd-namespace}-button;
+$message: #{$fd-namespace}-message;
```
Comment (InnaAtanasova) [inline src/message-box.scss:8]: Do you still need this?

File: src/message-box.scss [hunk 3]
```diff
@@ -111,18 +112,18 @@ $message-box-states: (
       }
 
       @include fd-rtl() {
-        .#{$block}__title::before {
+        .#{$message}-title::before {
```
Comment (InnaAtanasova) [inline src/message-box.scss:115]: I think this should be `#{$block}__title `.

File: src/message-box.scss [hunk 4]
```diff
@@ -111,18 +112,18 @@ $message-box-states: (
...
+  }
 
-      &.#{$block}--no-icon {
-        .#{$block}__title::before {
-          content: '';
-          margin: 0;
-        }
-      }
+  &--no-icon {
+    .#{$message}__title::before {
```
Comment (InnaAtanasova) [inline src/message-box.scss:124]: I think this should be `#{$block}__title `.

File: stories/message-box/__snapshots__/message-box.stories.storyshot [hunk 5]
```diff
@@ -137,7 +137,7 @@ exports[`Storyshots Components/Message Box No icon 1`] = `
             
                     
             <h2
-              class="fd-title fd-title--h5"
+              class="fd-title fd-title--h5 fd-message__title"
```
Comment (InnaAtanasova) [inline stories/message-box/__snapshots__/message-box.stories.storyshot:140]: I think you haven't updated the storyshot as this should be ` fd-message-box__title`, not `fd-message__title`

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
Please check again if you are referencing the correct issue in the PR template.

File: src/message-box.scss [hunk 6]
```diff
@@ -100,7 +101,7 @@ $message-box-states: (
       }
 
       .#{$block}__title {
-        @include fd-icon-base () {
+        @include fd-icon-base ("before") {
```
Comment (JKMarkowski) [inline src/message-box.scss:104]: It can be removed, `before` is set by default

**Review by JKMarkowski — COMMENTED** [review]
Let's apply Inna's comments, I added something else. Can you also run `npm run test:update` to refresh storyshots? Rest LGTM

File: src/message-box.scss (hunk 6 above)
Comment (salarenko) [inline src/message-box.scss:104]: Hey, some time ago we agreed no to create icons this way. Icons should be moved to separate `<i>`

**Review by salarenko — CHANGES_REQUESTED** [review]

**Comment by InnaAtanasova** [comment]
Again, this PR is not fixing the issue linked in the PR template.

File: stories/message-box/message-box.stories.js [hunk 7]
```diff
@@ -248,11 +254,12 @@ Information | \`fd-message-box--information\` | Information messages provide inf
 
 export const noIcon = () =>
     `
-<div class="fd-message-box-docs-static fd-message-box fd-message-box--no-icon fd-message-box--success fd-message-box--active">
+<div class="fd-message-box-docs-static fd-message-box fd-message-box--success fd-message-box--no-icon fd-message-box--active">
     <section class="fd-message-box__content">
         <header class="fd-bar fd-bar--header fd-message-box__header">
             <div class="fd-bar__left">
                 <div class="fd-bar__element">
+                    <i class="fd-message-box__title"></i>
```
Comment (JKMarkowski) [inline stories/message-box/message-box.stories.js:262]: I don't think that `__title` is proper name for icon there, can you change it to basic `__icon`?

Comment (stefanoScalzo) [reply stories/message-box/message-box.stories.js]: fixed

File: src/message-box.scss (hunk 6 above)
Comment (stefanoScalzo) [reply src/message-box.scss]: fixed

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Why can't I highlight text or click the buttons in these examples?  And the button text isn't centered

Screen Shot 2020-11-30 at 6 42 10 PM

**Comment by JKMarkowski** [comment]
@stefanoScalzo could you address @mikerodonnell89 Issue there a well?
examples wrappers have `position: static;`, which are supposed to be `relative`.

**Comment by stefanoScalzo** [comment]
> examples wrappers have `position: static;`, which are supposed to be `relative`.

Can we tackle this in another PR? @JKMarkowski @mikerodonnell89

**Comment by stefanoScalzo** [comment]
> > examples wrappers have `position: static;`, which are supposed to be `relative`.
Screen Shot 2020-12-01 at 2 56 10 PM
We use static or else when you scroll with storybook it produces this effect

@JKMarkowski

**Comment by InnaAtanasova** [comment]
Please rebase with latest master and check if the issues mentioned above are still relevant. Thanks

**Review by salarenko — APPROVED** [review]

**Comment by stefanoScalzo** [comment]
> Please rebase with latest master and check if the issues mentioned above are still relevant. Thanks

I've rebased and fix comments. Master has an issue with failing grid list test

**Review by mikerodonnell89 — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
Closing in favour of https://github.com/SAP/fundamental-styles/pull/1945

File: stories/message-box/message-box.stories.js [hunk 8]
```diff
@@ -51,12 +52,13 @@ Note: Include two action buttons in the message box when the user's decision is
 
 const messageBoxHeight = 200;
 
-export const structure = () => `<div class="fd-message-box-docs-static fd-message-box fd-message-box--active">
```
Comment (JKMarkowski) [inline stories/message-box/message-box.stories.js:54]: Don't remove `fd-message-box-docs-static`, It's needed there. I would add `z-index: 1` to this class, not to the lib

File: src/message-box.scss [hunk 9]
```diff
@@ -27,14 +27,25 @@ $message-box-states: (
 .#{$block} {
   @include fd-reset();
 
+  position: relative;
+  z-index: 1;
```
Comment (JKMarkowski) [inline src/message-box.scss:31]: I don't think if it's good idea to put it there, can you add it to `fd-message-box-docs-static` this class?

**Review by JKMarkowski — CHANGES_REQUESTED** [review]
Examples LGTM, but check my comments about the code

**Pull request closed without merging**